service/src/event/migration.rs

```diff
@@ -219,16 +219,29 @@ impl<'a, S: BlockStore> Migrator<'a, S> {
             .push(event_builder.build(&self.network, event).await?);
         Ok(())
     }
-    fn is_tile_doc(&self, data: &[u8]) -> bool {
+    fn is_tile_doc_init(&self, data: &[u8]) -> bool {
         // Attempt to decode the payload as a loose TileDocument.
         // If we succeed produce a meaningful error message.
+        // An init event will always have a `header` field.
         #[allow(dead_code)]
         #[derive(Debug, Deserialize)]
-        struct TileDocPayload {
+        struct TileDocInitPayload {
+            header: Ipld,
+        }
+        serde_ipld_dagcbor::from_slice::<TileDocInitPayload>(data).is_ok()
+    }
+    fn is_tile_doc_data(&self, data: &[u8]) -> bool {
+        // Attempt to decode the payload as a loose TileDocument.
+        // If we succeed produce a meaningful error message.
```

Comment:
does the comment `// If we succeed produce a meaningful error message` still apply here and above?

Reply:
Removed comment, thanks